app/controllers/madmin/resource_controller.rb

```
@@ -91,13 +98,27 @@ def new_resource_params
     def change_polymorphic(data)
       return data unless data.is_a?(ActionController::Parameters) && data[:type]
 
-      if data[:type] == "polymorphic"
-        GlobalID::Locator.locate(data[:value])
-      else
-        raise "Unrecognised param data: #{data.inspect}"
+      raise "Unrecognised param data: #{data.inspect}" unless data[:type] == "polymorphic"
+
+      GlobalID::Locator.locate(data[:value])
+    end
+
+    def parse_json_attribute(key, value)
+      return value unless json_attribute?(key)
+      return value unless value.is_a?(String) && value.present?
+
+      begin
+        JSON.parse(value)
+      rescue JSON::ParserError
+        value
```

Shouldn't we let this raise an error since the field should be valid JSON?